Add sidebar, accept new fortune input

AlbertXDing opened the pull request:
Sidebar with eight previously generated prompts. If the user clicks on one of them, a prompt appears where the user can change the prompts.

AlbertXDing committed:
Added prompts and gave users ability to change prompts

Sidebar with eight previously generated prompts. If the user clicks on one of them, a prompt appears where the user can change the prompts.

AlbertXDing committed:
Update main.js

Fixed indentation(dropped my water bottle on my keyboard)

phentos commented on assets/scripts/main.js:
```diff
@@ -1,5 +1,21 @@
-
 document.addEventListener('DOMContentLoaded', () => {
-    // init actions
+	const buttons = document.querySelectorAll('button');
```
are you sure you want to grab every button everywhere?

phentos commented on assets/scripts/main.js:
```diff
@@ -1,5 +1,21 @@
+			});
+
+		});
+	});
```
move to their own function(s), this is way too many things for the page load function

for example, create a new function called `activateSidebarHandler` to call from the page load

also bear in mind that the sidebar has nothing to do with the popup to enter a new fortune, so they should be in separate functions (ie, the sidebar listener calls the popup function to get the new value from the user)

phentos commented on assets/scripts/main.js:
```diff
@@ -1,5 +1,21 @@
+	const buttons = document.querySelectorAll('button');
+    const editBox = document.getElementById('editBox');
+    const save = document.querySelector('.save');
+	const userInput = document.getElementById('userInput');
```
descriptive names are really important:
- `editBox` -- this is meant to be the input field for the user to edit a fortune, right? so it should be named to reflect that, something like `userInputFortuneEditBox`
- `save` -- I can't tell what this is meant to save by reading it, we should be able to tell by the name. Perhaps use something like `userInputFortuneSubmitButton`
- `userInput` sorta works, but it would be nice if we could tell what the element is, especially since it's part of the input overlay

phentos commented on index.html:
```diff
@@ -1,12 +1,25 @@
+		<button class="button5">It is certain</button>
+		<button class="button6">Don't Count on it</button>
+		<button class="button7">Better not tell you now</button>
+		<button class="button8">As I see it, yes</button>
```
these should be part of a custom element / named container /etc rather than floating in `body`

phentos commented on index.html:
```diff
@@ -1,12 +1,25 @@
 <!DOCTYPE html>
-<html lang="en">
```
why did you remove `lang="en"`?

phentos commented on index.html:
```diff
@@ -1,12 +1,25 @@
+		<div id="editBox" style="display: none;">
+			<input type="text" id="userInput" class="text-input" placeholder="Enter your text">
+			<button class = "save">Save</button>
+		</div>
```
same deal with earlier comment, better names would be good

phentos reviewed (changes requested):
Thanks for the initiative porting Figma! I definitely wasn't looking forward to it myself lol.  

Here's my rationale for most of the changes requested: we want to keep things manageable as we add more functionality, so lets keep things as clean and compartmentalized as we can before merging them. With that in mind, we want to minimize side effects (for example, manipulating every button on the page instead of just the ones you need) and actions per function (a function should do 1 or 2 things; if it needs to do more, it should be multiple functions).

phentos committed:
sync from main

AlbertXDing committed:
Fixes Suggestions on pull request

Seperated main.js into individual functions, added better names, organized the buttons, gave elements in the sidebar their own class to distinguish from other buttons.

AlbertXDing commented:
Requested changes have been fixed in commit [de41dfc](https://github.com/cse110-sp23-group1/Origami-Fortune-Teller/pull/15/commits/de41dfc2d617d665095f3692a4f813b2fa01dc17)

phentos commented on assets/scripts/main.js:
```diff
@@ -1,21 +1,29 @@
-});
+    editFortuneBox.style.display = 'block';
+
+    saveButton.addEventListener('click', () => {
+        button.textContent = userInput.value;
+        editFortuneBox.style.display = 'none';
+    });
```
We don't replace `textContent` before or after the user gives a value, meaning every time they edit it will be prefilled with the last thing they submitted. Imo, it should have the fortune they are currently editing instead -- warrants discussion and an ADR @adarsh249

phentos replied:
moved to #23

phentos reviewed (approved):
Awesome work! Huge improvement, much more readable, much more expandable. Great job by @AlbertXDing
